codegen/scripts/add-version-file.js

```diff
@@ -0,0 +1,14 @@
+import fs from 'fs';
+import { fileURLToPath } from 'url';
+import path from 'path';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
```

Comment:
In ES Modules ``__dirname`` is not a global anymore, there might be better approaches. So far this made sense for those comming from CJS.